README.md

```diff
@@ -32,10 +49,97 @@ const plugins = [
+
+
+
+## Storefront Setup
+
+This guide will take you through how to set up Paystack payments in your Medusa storefront. It includes the steps necessary when using one of Medusa’s official storefronts as well as your own custom React-based storefront.
+
+### Storefront Prerequisites
+
+All storefronts require that you obtain your Paystack’s Public Key. You can retrieve it from your Paystack’s dashboard.
+
+### Add to Next.js Storefront
+
+Medusa has a Next.js storefront that you can easily use with your Medusa server. If you don’t have the storefront installed, you can follow [this quickstart guide](../starters/nextjs-medusa-starter).
+
+In your `.env.local` file (or the file you’re using for your environment variables), add the following variable:
+
+```bash
+NEXT_PUBLIC_PAYSTACK_PUBLIC_KEY=<PAYSTACK_PUBLIC_KEY>
+```
+
+Make sure to replace `<PAYSTACK_PUBLIC_KEY>` with your Paystack Public Key.
+
+Now, if you run your Medusa server and your storefront, on checkout you’ll be able to use Paystack.
+
+![Next.js Paystack Form](https://res.cloudinary.com/femakin/image/upload/v1666371607/Screenshot_2022-10-21_at_17.57.32_fx2nnq.png)
+
+
+
+### Add to Gatsby Storefront
+
+Medusa also has a Gatsby storefront that you can use as your ecommerce store. If you don’t have the storefront installed, you can follow [this quickstart guide](../starters/gatsby-medusa-starter).
+
+In your `.env.development` file (or the file you’re using for your environment variables) add the following variable with the value set to the Public Key:
+
+```jsx
+GATSBY_PAYSTACK_KEY=pk_
+```
+
+:::note
+
+You might find this environment variable already available so you can just replace its value with your Publishable Key.
+
+
+### Add to Custom Storefront
+
+This section guides you to add Paystack into a React-based framework. The instructions are general instructions that you can use in your storefront.
+
+The integration with Paystack must have the following workflow:
+
+1. During checkout when the user reaches the payment section, you should create payment sessions. This will initialize the payment_sessions array in the cart object received. The payment_sessions list contains an array of available payment providers.
+
+2. If Paystack is available as a payment provider, you should select Paystack as the payment session for the current cart. This will initialize the payment_session object in the cart object to include data related to Paystack and the current payment session. The payment intent and client secret are included here.
+
+3. After the user enters their card details and submits the form, confirm the payment with Paystack.
+
+4. If the payment is successful, complete the order in Medusa. Otherwise show an error.
+
+In your storefront, you need to install the [React Paystack components library](https://www.npmjs.com/package/react-paystack) and the [Medusa JS Client library](https://www.npmjs.com/package/@medusajs/medusa-js) with the foloowing codes:
```

Comment:
I don't think this should be React specific, what do you think?
Maybe just link to the accept payment docs https://paystack.com/docs/payments/accept-payments with a note about using the reference the plugin provides and the callback calling Medusa's `completeCheckout` function